src/components/balances/AssetsTable/useHideAssets.ts

```diff
@@ -0,0 +1,90 @@
+import useChainId from '@/hooks/useChainId'
+import useHiddenTokens from '@/hooks/useHiddenTokens'
+import { useAppDispatch } from '@/store'
+import { setHiddenTokensForChain } from '@/store/settingsSlice'
+import { useCallback, useState } from 'react'
+
+// This is the default for MUI Collapse
+export const COLLAPSE_TIMEOUT_MS = 300
+
+export const useHideAssets = (closeDialog: () => void) => {
+  const dispatch = useAppDispatch()
+  const chainId = useChainId()
+
+  const [assetsToHide, setAssetsToHide] = useState<string[]>([])
+  const [assetsToUnhide, setAssetsToUnhide] = useState<string[]>([])
+  const [hidingAsset, setHidingAsset] = useState<string>()
+  const hiddenAssets = useHiddenTokens()
+
+  const toggleAsset = useCallback(
+    (address: string) => {
+      if (assetsToHide.includes(address)) {
+        const newAssetsToHide = [...assetsToHide]
+        newAssetsToHide.splice(newAssetsToHide.indexOf(address), 1)
+        setAssetsToHide(newAssetsToHide)
+        return
+      }
+
+      if (assetsToUnhide.includes(address)) {
+        const newAssetsToUnhide = [...assetsToUnhide]
+        newAssetsToUnhide.splice(newAssetsToUnhide.indexOf(address), 1)
+        setAssetsToUnhide(newAssetsToUnhide)
+        return
+      }
+
+      const assetIsHidden = hiddenAssets.includes(address)
+      if (!assetIsHidden) {
+        setAssetsToHide([...assetsToHide, address])
```

Comment:
And for this, IMHO, concat is better. `setAssetsToHide((items) => items.concat(address))`.